File: assign3.py
class Node:
    def __init__(self, name):
        self.name = name
        self.children = []

def printNodes(node, depth):
    print("  " * depth + "- " + node.name)
    for child in node.children:
        printNodes(child, depth+1)

book = Node("Book")
chapter1 = Node("Chapter 1")
chapter2 = Node("Chapter 2")
book.children.append(chapter1)
book.children.append(chapter2)
section1_1 = Node("Section 1.1")
section1_2 = Node("Section 1.2")
chapter1.children.append(section1_1)
chapter1.children.append(section1_2)
section2_1 = Node("Section 2.1")
section2_2 = Node("Section 2.2")
chapter2.children.append(section2_1)
chapter2.children.append(section2_2)
subSection1_1_1 = Node("Sub-section 1.1.1")
subSection1_1_2 = Node("Sub-section 1.1.2")
subSection2_2_1 = Node("Sub-section 2.2.1")
subSection2_2_2 = Node("Sub-section 2.2.2")
section1_1.children.append(subSection1_1_2)
section1_1.children.append(subSection1_1_1)
section2_2.children.append(subSection2_2_1)
section2_2.children.append(subSection2_2_2)

printNodes(book, 0)

File: assign3.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Node
{
public:
    string name;
    vector<Node *> children;

    Node(string name)
    {
        this->name = name;
    }
};

void printNodes(Node *node, int depth)
{
    for (int i = 0; i < depth; i++)
    {
        cout << "  ";
    }
    cout << "- " << node->name << endl;
    for (Node *child : node->children)
    {
        printNodes(child, depth + 1);
    }
}

int main()
{
    // Create the tree structure
    Node *book = new Node("Book");
    Node *chapter1 = new Node("Chapter 1");
    Node *chapter2 = new Node("Chapter 2");
    book->children.push_back(chapter1);
    book->children.push_back(chapter2);
    Node *section1_1 = new Node("Section 1.1");
    Node *section1_2 = new Node("Section 1.2");
    chapter1->children.push_back(section1_1);
    chapter1->children.push_back(section1_2);
    Node *section2_1 = new Node("Section 2.1");
    chapter2->children.push_back(section2_1);
    Node *section2_2 = new Node("Section 2.2");
    chapter2->children.push_back(section2_2);
    Node *subSection1_1_1 = new Node("Sub-section 1.1.1");
    section1_1->children.push_back(subSection1_1_1);
    Node *subSection1_1_2 = new Node("Sub-section 1.1.2");
    section1_1->children.push_back(subSection1_1_2);
    Node *subSection2_2_1 = new Node("Sub-section 2.2.1");
    section2_2->children.push_back(subSection2_2_1);
    Node *subSection2_2_2 = new Node("Sub-section 2.2.2");
    section2_2->children.push_back(subSection2_2_2);

    // Print the nodes of the tree
    printNodes(book, 0);

    return 0;
}
